sympy/stats/drv_types.py

```diff
@@ -97,31 +302,42 @@ def Poisson(name, lamda):
     return rv(name, PoissonDistribution, lamda)
 
 
-class GeometricDistribution(SingleDiscreteDistribution):
-    _argnames = ('p',)
+#-------------------------------------------------------------------------------
+# Yule-Simon distribution ------------------------------------------------------------
+
+class YuleSimonDistribution(SingleDiscreteDistribution):
+    _argnames = ('rho',)
     set = S.Naturals
 
     @staticmethod
-    def check(p):
-        _value_check(And(0 < p, p<=1), "p must be between 0 and 1")
+    def check(rho):
+        _value_check(rho > 0, 'rho should be positive')
 
     def pdf(self, k):
-        return (1 - self.p)**(k - 1) * self.p
+        rho = self.rho
+        return rho * beta(k, rho + 1)
 
+    def _cdf(self, x):
+        return 1 - x * beta(x, self.rho + 1)
```

Comment:
The CDF is defined for all real inputs x, and so it should return correct values for all reals. With this formula, cdf(X)(0) is 1. I think 
```
Piecewise((1 - floor(x) * beta(floor(x), self.rho + 1), x >= 1), (0, True))
```
would be correct.